### PR TITLE
added mtev_confstr_parse_time_gm routine.

### PR DESCRIPTION
**Pull request by aidancully**
parses rfc3339 datespecs.

also, for consistency, changed existing mtev_confstr to return
MTEV_CONFSTR_PARSE result codes.

### CODE REVIEW

File: src/utils/mtev_confstr.c [hunk 1]
```diff
@@ -122,13 +123,13 @@ int
 mtev_confstr_parse_boolean(const char *input, mtev_boolean *output) {
   if(!strcasecmp(input, "yes") || !strcasecmp(input, "true") || !strcasecmp(input, "on")) {
```
Comment (rileyberton) [inline src/utils/mtev_confstr.c]: Maybe also support "1"?

File: src/utils/mtev_confstr.c [hunk 2]
```diff
@@ -122,13 +123,13 @@ int
...
     *output = mtev_true;
-    return 1;
+    return MTEV_CONFSTR_PARSE_SUCCESS;
   }
   if(!strcasecmp(input, "no") || !strcasecmp(input, "false") || !strcasecmp(input, "off")) {
```
Comment (rileyberton) [inline src/utils/mtev_confstr.c]: Maybe also support "0"?

File: src/utils/mtev_confstr.c (hunk 1 above)
Comment (aidancully) [inline src/utils/mtev_confstr.c]: Maybe? but I'd put that in a different PR. For what it's worth, those strings were mostly copied from [reconnoiter](https://github.com/circonus-labs/reconnoiter/blob/master/src/modules/httptrap.c#L111).

**Comment by rileyberton** [comment]
Looks good to me.

File: src/utils/mtev_confstr.c (hunk 1 above)
Comment (postwait) [inline src/utils/mtev_confstr.c]: Reconnoiter and thus libmtev use on/true and off/ false. If add 1/0, I
agree it should be a separate and sweeping commit.
